### PR TITLE
Bridge tokens by redirecting to Portfolio from wallet overview page

### PR DESCRIPTION
**Pull request by micaelae**
## Explanation

This pull request adds a Bridge button to the wallet overview that opens the [MetaMask Portfolio Dapp bridge experience](https://portfolio.metamask.io/bridge) in a new tab when clicked.

## Changes
* Add new Bridge button and icon to eth-overview component
* Move Portfolio button next to displayed balance
* Add new `renderBackgroundColor` prop to IconButton to style Portfolio IconButton

### Before

Screenshot 2023-03-01 at 3 19 05 PM

### After

Screenshot 2023-03-03 at 10 51 31 AM

https://user-images.githubusercontent.com/100321200/222308227-981ecd35-e876-4754-b0ab-5bf793afd764.mov

## Manual Testing Steps
1. Open the extension
2. Click on the Bridge button
3. Verify that you are redirected to the MetaMask Portfolio Dapp bridge page
4. Open the extension
5. Click on the Portfolio button
6. Verify that you are redirected to the MetaMask Portfolio Dapp home page

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [X] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by micaelae** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/images/icons/bridge.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
+  <path d="m74 284c-21 7-36 26-36 49 0 28 23 51 52 51 28 0 51-23 51-51 0-20-12-38-28-46 14-65 73-114 143-114 47 0 89 22 116 56 12-6 25-10 39-11-33-50-90-84-155-84-90 0-165 65-182 150z m400 49c0 28-23 51-52 51-28 0-51-23-51-51 0-28 23-51 51-51 29 0 52 23 52 51z"/>
+</svg>
```
Comment (georgewrmarshall) [inline app/images/icons/bridge.svg:3]: Nice work! not sure if you need a line break here after the closing tag but GitHub is complaining.

Screenshot 2023-03-01 at 3 33 58 PM

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Just need to fix failing tests.
- checked storybook
- pulled branch and checked local build

Screenshot 2023-03-01 at 3 40 23 PM
Screenshot 2023-03-01 at 3 40 41 PM

**Comment by kevinghim** [comment]
To be clear, the agreed upon changes have 4 icons: buy, send, swap, bridge on the global actions row with an entry for Portfolio next to the account balance, correct @jakehaugen?
image (4)

File: app/images/icons/bridge.svg (hunk 1 above)
Comment (micaelae) [reply app/images/icons/bridge.svg]: Thanks! I just added a newline to the end to address this

**Comment by jakehaugen** [comment]
@kevinghim, correct -- 4 buttons.

This would be a more accurate mock applied to the current UI (with the Bridge icon added of course).

bridge-button

**Comment by micaelae** [comment]
I implemented the changes to move the Portfolio button next to the balance. This is what the updated home page looks like:

Screenshot 2023-03-01 at 5 26 35 PM

**Comment by BZahory** [comment]
Should we vertically center the portfolio icon? It looks a little off sitting at the bottom of the row like that.

**Comment by jakehaugen** [comment]
> Should we vertically center the portfolio icon? It looks a little off sitting at the bottom of the row like that.

Good call, in the future looking designs we had the balance font size bigger and bottom aligned looked nice. But at the current size I agree that vertically centered is a good move. Attached a mock for a visual reference. 

bridge-button-vertical-center

**Comment by micaelae** [comment]
Updated the PR to vertically center the Portfolio button and added the new screenshot to the description

File: ui/components/app/wallet-overview/eth-overview.js [hunk 2]
```diff
@@ -88,6 +88,37 @@ const EthOverview = ({ className }) => {
...
+                      },
+                    },
+                    {
+                      contextPropsIntoEventProperties: [
+                        CONTEXT_PROPS.PAGE_TITLE,
+                      ],
+                    },
+                  );
+                }}
+              />
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/app/wallet-overview/eth-overview.js:121]: All our components accept `Box` props https://metamask.github.io/metamask-storybook/?path=/docs/components-ui-box--default-story

File: ui/components/ui/icon-button/icon-button.js [hunk 3]
```diff
@@ -40,5 +53,6 @@ IconButton.propTypes = {
   label: PropTypes.string.isRequired,
   tooltipRender: PropTypes.func,
   className: PropTypes.string,
+  renderBackgroundColor: PropTypes.bool,
   'data-testid': PropTypes.string,
 };
```
Comment (georgewrmarshall) [inline ui/components/ui/icon-button/icon-button.js:58]: Shouldn't need to update this component if you use`ButtonIcon`

File: ui/components/ui/icon-button/icon-button.scss [hunk 4]
```diff
@@ -24,6 +24,12 @@
...
 
+  &__no_background {
+    background: var(--color-background-default);
+    width: 20px;
+    height: 20px;
+  }
+
   &--disabled {
     opacity: 0.3;
     cursor: auto;
```
Comment (georgewrmarshall) [inline ui/components/ui/icon-button/icon-button.scss:35]: Can be removed

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Left some suggestions

File: ui/components/app/wallet-overview/eth-overview.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (micaelae) [reply ui/components/app/wallet-overview/eth-overview.js]: Thanks for catching this! I missed the console error and wasn't aware of IconButton. I updated the PR to use it

File: ui/components/app/wallet-overview/index.scss [hunk 5]
```diff
@@ -70,6 +71,14 @@
     color: var(--color-warning-default);
   }
 
+  &__portfolio {
+    position: absolute;
+    right: -32px;
+    height: 44.8px;
+    padding: auto 0 auto 0;
+    background: inherit;
+  }
```
[the remark this answers is not in the document]
Comment (micaelae) [reply ui/components/app/wallet-overview/index.scss]: Removed in latest revision

File: ui/components/app/wallet-overview/eth-overview.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/app/wallet-overview/eth-overview.js]: np! extension is a lot messier to work with I'm sure :D

File: ui/components/component-library/button-icon/button-icon.constants.js [hunk 6]
```diff
@@ -2,5 +2,6 @@ import { Size } from '../../../helpers/constants/design-system';
 
 export const BUTTON_ICON_SIZES = {
   SM: Size.SM,
+  MD: Size.MD,
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-icon/button-icon.constants.js:5]: We don't offer this in Figma so I'm doubtful this is the size used with the ButtonIcon button. Have reached out to @SaraCheikh to confirm

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. Made one more suggestion and need to confirm icon size. But we could ship after these changes if need be. Thanks for you patience 🙏

File: ui/components/component-library/button-icon/button-icon.constants.js (hunk 6 above)
Comment (georgewrmarshall) [reply ui/components/component-library/button-icon/button-icon.constants.js:5]: This is what it looks like different icon sizes
Icon Sm

Screenshot 2023-03-03 at 9 27 35 AM

Icon Md (doesn't exist currently)

Screenshot 2023-03-03 at 9 27 46 AM

Icon Lg

Screenshot 2023-03-03 at 9 28 02 AM

Comment (georgewrmarshall) [reply ui/components/component-library/button-icon/button-icon.constants.js:5]: As per slack conversation use `size={BUTTON_ICON_SIZES.LG}`

File: ui/components/app/wallet-overview/eth-overview.js [hunk 7]
```diff
@@ -88,6 +93,40 @@ const EthOverview = ({ className }) => {
               {balanceIsCached ? (
                 <span className="eth-overview__cached-star">*</span>
               ) : null}
+              <ButtonIcon
+                data-testid="home__portfolio-site"
+                color={IconColor.primaryDefault}
+                iconName={ICON_NAMES.DIAGRAM}
+                ariaLabel={t('portfolio')}
+                size={BUTTON_ICON_SIZES.MD}
```
Comment (georgewrmarshall) [inline ui/components/app/wallet-overview/eth-overview.js:101]: As per slack conversation
```suggestion
                size={BUTTON_ICON_SIZES.LG}
```

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Nice work 💯 🚀

**Review by darkwing — DISMISSED** [review]

File: coverage-targets.js [hunk 8]
```diff
@@ -6,10 +6,10 @@
...
   global: {
-    lines: 64.39,
-    branches: 53.03,
-    statements: 63.66,
-    functions: 56.67,
+    lines: 68.63,
+    branches: 47.48,
+    statements: 66.9,
+    functions: 62.48,
   },
```
Comment (georgewrmarshall) [inline coverage-targets.js:13]: Was getting `Error: Coverage EXCEEDS threshold for global and must be bumped`

Screenshot 2023-03-03 at 11 29 33 AM

File: ui/components/ui/icon-button/icon-button.stories.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  args: {
+    Icon: <Icon name={ICON_NAMES.CARD} color={IconColor.primaryInverse} />,
+    label: 'Buy',
+  },
+};
+
+export const DefaultStory = (args) => <IconButton {...args} />;
+
+DefaultStory.storyName = 'Default';
```
Comment (georgewrmarshall) [inline ui/components/ui/icon-button/icon-button.stories.js:35]: Adding IconButton story for easier future visual updates

Screenshot 2023-03-05 at 5 43 10 PM

**Review by georgewrmarshall — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**